pkg/framework/machines.go

```diff
@@ -159,5 +159,5 @@ func WaitForMachinesDeleted(c runtimeclient.Client, machines ...*machinev1.Machi
 		}
 
 		return true // Everything was deleted.
-	}, WaitLong, RetryMedium).Should(BeTrue())
+	}, WaitLong, RetryMedium).Should(BeTrue(), "error encounted while deleting Machines.")
```

Comment:
Maybe this instead? Since this is just waiting?
```suggestion
	}, WaitLong, RetryMedium).Should(BeTrue(), "error encountered while Machines to be deleted.")
```